multimodal/src/autogluon/multimodal/optimization/lit_mmdet.py

```diff
@@ -0,0 +1,317 @@
+                    {
+                        'filename': 'data/VOCdevkit/VOC2007/JPEGImages/000001.jpg',
+                        'ori_filename': 'JPEGImages/000001.jpg',
+                        'ori_shape': (500, 353, 3),
+                        ...
+                    },
+                    ...(batch size times)
+                ],
+                "gt_bboxes": [
+                    torch.Tensor, Size: [# objects in image, 4],
+                    ...(batch size times)
+                ],
+                "gt_labels": [
+                    torch.Tensor, Size: [# objects in image],
+                    ...(batch size times)
+                ],
+            }
+            ```
+        """
+        losses = self.model.forward_train(
+            img=img,
+            img_metas=img_metas,
+            gt_bboxes=gt_bboxes,
+            gt_labels=gt_labels,
+            *args,
+            **kwargs,
+        )
+        return self.parse_losses(losses)
+
+    def parse_losses(self, losses):
+        # `_parse_losses`: https://github.com/open-mmlab/mmdetection/blob/
+        # 56e42e72cdf516bebb676e586f408b98f854d84c/mmdet/models/detectors/base.py#L176
+        loss, log_vars = self.model._parse_losses(losses)
+        return loss, log_vars
+
+    def log_step_results(self, losses):
+        map_loss_names = {
+            "loss": "total_loss",
+            "acc": "classification-accuracy",
+            "loss_rpn_cls": "loss/rpn_cls",
+            "loss_rpn_bbox": "loss/rpn_bbox",
+            "loss_bbox": "loss/bbox_reg",
+            "loss_cls": "loss/classification",
+        }
+        for key in losses:
+            # map metric names same name with epoch-wise metrics defined in:
+            # `configs/vision/object-detection/mmdet/mmdet-base.yaml`
+            loss_name = map_loss_names.get(key, key)
+            self.log(f"step/{loss_name}", losses[key])
+
+    def training_step(self, batch, batch_idx):
+        loss, log_vars = self._training_step(batch, batch_idx)
+        return loss
+
+    def validation_step(self, batch, batch_idx, dataloader_idx=0):
+        self.evaluate(batch, "val")
+
+    def validation_epoch_end(self, validation_step_outputs):
+        # TODO: add mAP/mAR_per_class
+        mAPs = {"val_" + k: v for k, v in self.validation_metric.compute().items()}
+        self.print(mAPs)
+        self.log_dict(mAPs, sync_dist=True)
+        self.validation_metric.reset()
+
+    def test_step(self, batch, batch_idx, dataloader_idx=0):
+        raise NotImplementedError("test with lit_mmdet is not implemented yet.")
+        res = self.evaluate(batch, "test")
+        return res
+
+    def predict_step(self, batch, batch_idx, dataloader_idx=0):
+        pred = self._predict_step(batch, batch_idx)
+        if "mmdet_image_label" in batch:
+            return {"bbox": pred, "label": batch["mmdet_image_label"]}
+        else:
+            return {"bbox": pred}
+
+    def configure_optimizers(self):
```

Comment:
Is this `configure_optimizers ` the same as the one in `lit_module`? If so, consider reusing it by inheritance or util.

Reply:
yes for now, but later we may want to freeze backbone and only train bboxhead, or using different lr for them.